File: problems/844-backspace-string-compare/main.js
const backspaceCompare = function(S, T) {
  const first = [];
  const second = [];

  for (let i = 0; i < S.length; i++) {
    if (S[i] !== '#') {
      first.push(S[i]);
    } else if (first.length) {
      first.pop();
    }
  }

  for (let i = 0; i < T.length; i++) {
    if (T[i] !== '#') {
      second.push(T[i]);
    } else if (second.length) {
      second.pop();
    }
  }

  return first.join('') === second.join('');
};

File: problems/844-backspace-string-compare/main.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        first = []
        second = []
        for i in range(0, len(S)):
          if (S[i] != '#'):
            first.append(S[i])
          elif len(first):
            first.pop()
        
        for i in range(0, len(T)):
          if (T[i] != '#'):
            second.append(T[i])
          elif len(second):
            second.pop()
        return "".join(first) == "".join(second)
